#**Mendeteksi Berita Palsu**

In [1]:
from google.colab import files
uploaded = files.upload()

Saving news.zip to news.zip


In [2]:
!unzip 'news.zip'

Archive:  news.zip
  inflating: news.csv                


###**Mengimpor Seluruh Library Yang Diperlukan**

In [3]:
import numpy as np
import pandas as pd 
import itertools
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.metrics import accuracy_score, confusion_matrix

###**Memasukkan Data Ke Dalam Data Frame dan Mendapatkan Series Label**

In [4]:
data = pd.read_csv('news.csv')

In [18]:
data.shape
data.head()

Unnamed: 0                                              title  \
6330        4490  State Department says it can't find emails fro...   
6331        8062  The ‘P’ in PBS Should Stand for ‘Plutocratic’ ...   
6332        8622  Anti-Trump Protesters Are Tools of the Oligarc...   
6333        4021  In Ethiopia, Obama seeks progress on peace, se...   
6334        4330  Jeb Bush Is Suddenly Attacking Trump. Here's W...   

                                                   text label  
6330  The State Department told the Republican Natio...  REAL  
6331  The ‘P’ in PBS Should Stand for ‘Plutocratic’ ...  FAKE  
6332   Anti-Trump Protesters Are Tools of the Oligar...  FAKE  
6333  ADDIS ABABA, Ethiopia —President Obama convene...  REAL  
6334  Jeb Bush Is Suddenly Attacking Trump. Here's W...  REAL

In [6]:
labels = data.label
labels.head()

0    FAKE
1    FAKE
2    REAL
3    FAKE
4    REAL
Name: label, dtype: object

### **Mengecek Missing Value Pada Label**

In [9]:
data[pd.isnull(data.label)]

Empty DataFrame
Columns: [Unnamed: 0, title, text, label]
Index: []

### **Machine Learning Modelling**

In [13]:
x_train, x_test, y_train, y_test = train_test_split(data['text'], labels, test_size = 0.2, random_state = 7)
#Memisahkan Dataset Menjadi Dataset Train dan Test

In [11]:
tfidf_vectorizer = TfidfVectorizer(stop_words='english', max_df = 0.7)
# Menginiliasasi TfidfVectorizer Dengan Stop Words English Language dan Maximun Document Frequency sebesar 0.7

In [14]:
tfidf_train = tfidf_vectorizer.fit_transform(x_train)
tfidf_test = tfidf_vectorizer.transform(x_test)
# Memasang dan Merubah Vectorizer Pada Dataset Train dan hanya merubah Vectorizer Pada Dataset Test

In [15]:
pac = PassiveAggressiveClassifier(max_iter = 50)
pac.fit(tfidf_train, y_train)
# Menginiliasasi PassiveAggresiveClassifer Dengan Pengulangan Makasimal Sebesar 50 dan Memasangkannya Pada tfidf_train & y_train

PassiveAggressiveClassifier(max_iter=50)

In [16]:
y_pred = pac.predict(tfidf_test)
score = accuracy_score (y_test, y_pred)
print(f'Accuracy : {round(score*100,3)}%')
# Mengetes Akurasi Dari Model Yang Telah Dibuat

Accuracy : 92.739%


In [17]:
confusion_matrix(y_test, y_pred, labels=['FAKE', 'REAL'])
#Membuat Confusion Matrix

array([[588,  50],
       [ 42, 587]])

## **Summary**
Dari Hasil Pembuatan Model, Kita Mendapatkan Akurasi Model Sebesar 92.739% dan Mendapatkan Confusion Matrix Dimana 589 True Positives, 587 True Negatives, 42 False Positives, and 49 False Negatives.